client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -21,9 +21,9 @@ export const BaseDatePickerInput: FC<
   const [internalValue, setInternalValue] = useState<Date | null>(null);
 
   useEffect(() => {
-    // This sets the internal state from parent when first loading (i.e. when
-    // the internal date is still empty)
-    if (value && internalValue === null)
+    // This sets the internal state from parent on first load (when internal
+    // date is empty) or when value has changed and internal date needs updating
+    if (value && (internalValue !== value || internalValue === null))
```

Comment:
The only problem here is that when `value` is a Date object rather than a string, the `internalValue !== value` comparison will always return `true` (since they're references to different objects even if their value might be the same).

I'm not sure if this is actually a problem, but I do remember some infinite re-render loops caused by this kind of thing. Can you maybe use the `isEqual` check from `DateUtils` to compare Date objects?

Reply:
Oh, you also probably don't need the `||`, since if internalValue is `null`, then the `!=` would cover that case, right?

Reply:
So something like:
```ts
 if (value && ((typeof value === "string" && internalValue !== value ) || (!DateUtils.isEqual(internalValue, value)))
```

Not 100% sure haven't tested ;)

Reply:
[ba61977](https://github.com/openmsupply/open-msupply/pull/1922/commits/ba61977ebe2913a872d8abb1d9d76095fee0b896) Can I compare string? 👀

Reply:
We're just using `isEqual` straight out of the `date-fns` library, so probably shouldn't try and unpack it.

The string comparison should be fine, but you need to convert them both, as `value` could be a Date or a string too.
```
if (value && (internalValue?.toString() !== value?.toString()))
```

Reply:
Opps missed that, done (: